frontend/components/examples-page.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Meta from './meta';
+import Header from './header';
+import SuggestionAndSearch from './suggestion-and-search';
+import SearchResults from './search-results';
+import MarkupModal from './markup-modal';
+import Usage from './usage';
+import Footer from './footer';
+import { baseUri, longCache } from '../constants';
+
+export default class ExamplesPage extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      category: props.match.params.id,
+      query: null,
+      example: null,
+      searchReady: true,
+    };
+    this.searchTimeout = 0;
+    this.renderSearchResults = this.renderSearchResults.bind(this);
+    this.searchQueryChanged = this.searchQueryChanged.bind(this);
+  }
+
+  static propTypes = {
+    match: PropTypes.object.isRequired,
+  }
+
+  searchQueryChanged(query) {
+    this.setState({searchReady: false});
+    window.clearTimeout(this.searchTimeout);
```

Comment:
Could you add a comment here explaining the behavior? If I understand correctly, it's to wait until after the user stops typing, and then run the search, and to show the "searching" message while they're waiting.